Library/Formula/metashell.rb

```diff
@@ -10,20 +11,21 @@ class Metashell < Formula
     sha1 "76874d93ced1815ccfcbddb6da75e8c56aff44e4" => :mountain_lion
   end
 
-  depends_on "cmake" => :build
-
   needs :cxx11
 
   def install
     ENV.cxx11
     # Build internal Clang
-    mkdir "templight/build" do
+    mkdir "3rd/templight/build" do
       system "cmake", "../llvm", "-DLIBCLANG_BUILD_STATIC=ON", *std_cmake_args
       system "make", "clang"
       system "make", "libclang"
       system "make", "libclang_static"
+      system "make", "templight"
     end
 
+    system "tools/clang_default_path", "--gcc=clang", ">", "lib/core/extra_sysinclude.hpp"
```

Comment:
I have a couple of proposals on how this could be solved:
* Somehow force brew to use a newer compiler. I'm not 100% sure if it will solve the issue, or if it is possible to do at all with brew.
* Don't support Mountain Lion. I don't really like this option.
* Apply the following patch before compiling for Mountain Lion:

```diff
diff --git a/3rd/templight/llvm/tools/clang/tools/templight/TemplightDebugger.cpp b/3rd/templight/llvm/tools/clang/tools/templight/TemplightDebugger.cpp
index 7a5a2d3..c60d7de 100644
--- a/3rd/templight/llvm/tools/clang/tools/templight/TemplightDebugger.cpp
+++ b/3rd/templight/llvm/tools/clang/tools/templight/TemplightDebugger.cpp
@@ -672,6 +672,7 @@ public:
   };

   void processInputs() {
+#if 0
     std::string user_in;
     while(true) {
       llvm::outs() << "(tdb) ";
@@ -958,6 +959,7 @@ public:
       }

     };
+#endif
   };

   void printRawEntry(const TemplateDebuggerEntry &Entry) {
```

With the following comment:
```ruby
# This patch is required because Mountain Lion uses an old compiler which breaks
# compiling some Templight code. The patch comments out unused parts of Templight,
# so patched version is functionally equivalent. This error should be fixed in
# the next release of Metashell.
```

Hopefully this would eliminate all the (unused) code which causes the compilation issues with a minimal patch.